packages/core/components/DataTable/helpers/getChartCellValue.ts

```diff
@@ -31,16 +31,16 @@ export const getChartCellValue = (row: string, column: string, config: TableConf
   let cellValue // placeholder for formatting below
   let labelValue = rowObj[column] // just raw X axis string
   if (column === config.xAxis?.dataKey) {
-    // not the prettiest, but helper functions work nicely here.
-    cellValue =
-      config.xAxis?.type === 'date'
-        ? formatDate(
-            config.table?.dateDisplayFormat || config.xAxis?.dateDisplayFormat,
-            parseDate(config.xAxis?.dateParseFormat, labelValue)
-          )
-        : labelValue
-    cellValue =
-      config.xAxis?.type === 'continuous' ? formatNumber(runtimeData[row][column], 'bottom', false, config) : labelValue
+    let { type, dateDisplayFormat } = config.xAxis || {}
+    let dateFormat = config.table?.dateDisplayFormat || dateDisplayFormat
+
+    if (type === 'date' || type === 'date-time') {
+      cellValue = formatDate(dateFormat, parseDate(dateFormat, labelValue))
+    } else if (type === 'continuous') {
+      cellValue = formatNumber(runtimeData[row][column], 'bottom', false, config)
+    } else {
+      cellValue = labelValue
+    }

```

Comment:
this logic works otherwise, just change those above consts since those aren't variable.